pkg/scalers/scaler.go

```diff
@@ -99,6 +99,9 @@ type ScalerConfig struct {
 	// ScalerIndex
 	ScalerIndex int
 
+	// ScalerUniqueKey for the scaler across KEDA. Useful to identify uniquely the scaler, eg: AWS credentials cache
+	ScalerUniqueKey string
```

Comment:
Nit, shouldn't we name this `TriggerUniqueKey` and rename `ScalerIndex` -> `TriggerIndex` to be consistent with the rest of the stuff. 

The confusion between a trigger and scaler happenned a long time ago.

Reply:
Done, but it's a huge change that can be a bit unrelated with this